### PR TITLE
Add camera submission for multiple resolution test (New)

### PR DESCRIPTION
**Pull request by fernando79513**
## Description

The camera_test.py script has been modified to store one image each time it runs the multiple_resolution test.
We have fixed some bugs, renamed some variables to increase clarity and added tests

## Resolved issues

https://warthogs.atlassian.net/browse/CHECKBOX-1369

## Documentation

N/A

### Run the tests
```
cd providers/base
manage.py test -u
```

### Test the behavior
Run the camera automated tests:
```
com.canonical.certification::camera-cert-automated
```
Check that the multiple-resolution test image has been created in the PLAINBOX_SESSION_SHARE directory

### CODE REVIEW

File: providers/base/bin/camera_test.py [hunk 1]
```diff
@@ -637,9 +667,15 @@ def add_device_parameter(parser):
...
         action="store_true",
-        help=("Don't display picture, " "just write the picture to a file"),
+        help=("Don't display picture, just write the picture to a file"),
     )
     resolutions_parser = subparsers.add_parser("resolutions")
+    resolutions_parser.add_argument(
+        "-o",
+        "--output",
+        default="",
+        help="Output directory to store the images",
```
Comment (Hook25) [inline providers/base/bin/camera_test.py:677]:
```suggestion
        help="Output directory to store a small debug image",
```

File: providers/base/tests/test_camera_test.py [hunk 2]
```diff
@@ -32,22 +38,406 @@ class CameraTestTests(unittest.TestCase):
...
+                "description": "fake",
+                "resolutions": [[640, 480]],
+            },
+        ]
+        expected_str = (
+            "Format: YUYV (YUYV)\n"
+            "Resolutions: 640x480,320x240\n"
+            "Format: fake (fake)\n"
+            "Resolutions: 640x480\n"
+        )
```
Comment (Hook25) [inline providers/base/tests/test_camera_test.py:59]:
```suggestion
        expected_str = textwrap.dedent(
            """
            Format: YUYV (YUYV)
            Resolutions: 640x480,320x240
            Format: fake (fake)
            Resolutions: 640x480
            """
        ).lstrip()
```

Comment (Hook25) [reply providers/base/tests/test_camera_test.py:59]: (check if it works/formats is ok!)

File: providers/base/tests/test_camera_test.py [hunk 3]
```diff
@@ -32,22 +38,406 @@ class CameraTestTests(unittest.TestCase):
...
+    #                 # EINVAL is the ioctl's way of telling us that there are no
+    #                 # more formats, so we ignore it
+    #                 if e.errno != errno.EINVAL:
+    #                     print(
+    #                         "Unable to determine supported framesizes "
+    #                         "(resolutions), this may be a driver issue."
+    #                     )
+    #         supported_format["resolutions"] = resolutions
+    #     return supported_formats_info
+
```
Comment (Hook25) [inline providers/base/tests/test_camera_test.py:270]: Oooups, was this forgotten?
```suggestion
```

File: providers/base/bin/camera_test.py [hunk 4]
```diff
@@ -388,51 +389,81 @@ def resolutions(self):
...
+        # the closest resolution to 640x480 as the target to have some
+        # uniformity in the output
+        target = 640
+        closest_resolution = min(
+            format["resolutions"], key=lambda x: abs(x[0] - target)
+        )
+        w, h = closest_resolution
+        device_name = device.split("/")[-1]
+        filepath = os.path.join(
+            output, "resolution_test_image_{}.jpg".format(device_name)
```
Comment (Hook25) [inline providers/base/bin/camera_test.py:452]: Minor: on re-run this is overwritten, maybe we should put a `_{n}.jpg` at the end (you can simply add a len(list(glob(...)))`, also, consider pathlib.

File: providers/base/units/camera/jobs.pxu [hunk 5]
```diff
@@ -102,11 +102,29 @@ _summary: Webcam multiple resolution capture test for {product_slug}
...
+template-unit: job
+plugin: attachment
+category_id: com.canonical.plainbox::camera
+id: camera/multiple-resolution-images-attachment_{name}
+flags: also-after-suspend
+_summary: Attach an image from the multiple resolution images test for {product_slug}
+estimated_duration: 1s
+after: camera/multiple-resolution-images_{name}
+command:
+  [ -f "$PLAINBOX_SESSION_SHARE"/resolution_test_image_{name}.jpg ] &&
```
Comment (Hook25) [inline providers/base/units/camera/jobs.pxu:122]: If you adopt the change (n at the end) to the `resolution_test_image_{name}_*.jpg`, consider updating this as well

File: providers/base/units/camera/jobs.pxu [hunk 6]
```diff
@@ -162,6 +182,25 @@ _description:
...
+plugin: attachment
+category_id: com.canonical.plainbox::camera
+id: camera/multiple-resolution-images-rpi-attachment_{name}
+flags: also-after-suspend
+_summary: Attach an image from the multiple resolution images test for {product_slug}
+estimated_duration: 1s
+after: camera/multiple-resolution-images-rpi_{name}
+requires: cpuinfo.platform == 'armv7l'
+command:
+  [ -f "$PLAINBOX_SESSION_SHARE"/picam_6.jpg ] &&
```
Comment (Hook25) [inline providers/base/units/camera/jobs.pxu:198]: Why is the name off/different here?

**Review by Hook25 — CHANGES_REQUESTED** [review]
Nice, a few minor changes/suggestions, see what you think about them

File: providers/base/tests/test_camera_test.py (hunk 2 above)
Comment (fernando79513) [reply providers/base/tests/test_camera_test.py:59]: Yes, with the `lstrip()` it works fine

File: providers/base/bin/camera_test.py (hunk 4 above)
Comment (Hook25) [reply providers/base/bin/camera_test.py:452]: After what we decided, this is indeed good enough. If we will ever need the history we will go back to this

File: providers/base/units/camera/jobs.pxu (hunk 6 above)
Comment (Hook25) [reply providers/base/units/camera/jobs.pxu:198]: oh ok, ty for clarifying this

**Review by Hook25 — APPROVED** [review]

**Pull request merged**